monai/data/box_utils.py

```diff
@@ -572,3 +572,261 @@ def convert_box_to_standard_mode(
             box_convert_mode(boxes=boxes, src_mode="xxyyzz", dst_mode="xyzxyz")
     """
     return convert_box_mode(boxes=boxes, src_mode=mode, dst_mode=StandardMode())
+
+
+def check_boxes(boxes: NdarrayOrTensor) -> bool:
+    """
+    This function checks whether the box size is non-negative.
+
+    Args:
+        boxes: bounding box, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+
+    Returns:
+        whether ``boxes`` is valid
+    """
+    spatial_dims = get_spatial_dims(boxes=boxes)
+    for axis in range(0, spatial_dims):
+        if (boxes[:, spatial_dims + axis] < boxes[:, axis]).sum() > 0:
+            return False
+    return True
+
+
+def box_area(boxes: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    This function computes the area of each box
+
+    Args:
+        boxes: bounding box, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+
+    Returns:
+        area of boxes, with size of (N,).
+    """
+
+    if not check_boxes(boxes):
+        raise ValueError("Given boxes has invalid values. The box size must be non-negative.")
+
+    spatial_dims = get_spatial_dims(boxes=boxes)
+
+    area = boxes[:, spatial_dims] - boxes[:, 0] + TO_REMOVE
+    for axis in range(1, spatial_dims):
+        area = area * (boxes[:, axis + spatial_dims] - boxes[:, axis] + TO_REMOVE)
+
+    # convert numpy to tensor if needed
+    area_t, *_ = convert_data_type(area, torch.Tensor)
+
+    # check if NaN or Inf, especially for half precision
+    if area_t.isnan().any() or area_t.isinf().any():
+        if area_t.dtype is torch.float16:
```

Comment:
Hi @wyli ,

Is this a common practice in detection work? I think it may cause unexpected error, for example if we have 1000 images and use float16 to represent the boxes, but the box area of last image is bigger than float16, then we need to retrain all the images.

Thanks.

Reply:
It is actually used in _box_inter_union() and I always convert it to float32 before sending it as input of box_area(). 
- I can put a comment in docstring like "Half precision is not recommended for this function as it may cause overflow, especially for 3D images."
- Or I could always force it to run with float32, but that will change the dtype of output and may cause unexpected data type conversion

Reply:
I am OK with both solutions.
Please update it from the user's experience angle.

Thanks.

Reply:
I would avoid automatic data type conversion, as it changes the following data type and is hard to find the cause for the users